### PR TITLE
Fix broken industrial collision detection

### PR DESCRIPTION
**Pull request by jrgnicho**
This PR addresses the issue described in #70 where the **industrial_collision_detection** package broke after [changes made to moveit collision library](https://github.com/ros-planning/moveit/issues/488) that introduced the ability to perform distance queries along with some other improvements.  Thus this fix involves the following changes:
- Removed the duplicate Distance query data structures from industrial_collision_detection
- Added rather unpolished implementations of the new purely virtual functions inherited from the CollisionRobot and CollisionWorld interfaces.
- The constrained_ik package no longer depends on **industrial_collision_detection**

As of now the **industrial_collision_detection** package is no longer a dependency to any other package, nevertheless I chose to keep it around just in case.

This PR replaces #71 and #76 

@Levi-Armstrong I had to modify constrained_ik in order for it to build so hopefully I didn't break anything but could you suggest a way to check?

### CODE REVIEW

**Comment by davetcoleman** [comment]
> nevertheless I chose to keep it around just in case.

Does not git history achieve this "just in case" scenario? It seems to me keeping it will only confuse future contributors / developers and cause more maintenance hassle

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 1]
```diff
@@ -176,9 +125,9 @@ namespace collision_detection
     double dist_threshold = cdata->req->distance_threshold;
     std::map<std::string, DistanceResultsData>::iterator it1, it2;
 
-    if (!cdata->req->global)
+    if (false /*!cdata->req->global*/)
```
Comment (davetcoleman) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp]: add comment for why you are disabling this

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 2]
```diff
@@ -203,32 +152,32 @@ namespace collision_detection
...
-      dist_result.link_name[0] = cd1->ptr.obj->id_;
-      dist_result.link_name[1] = cd2->ptr.obj->id_;
+      dist_result.link_names[0] = cd1->ptr.obj->id_;
+      dist_result.link_names[1] = cd2->ptr.obj->id_;
 
-      cdata->res->minimum_distance.update(dist_result);
+      cdata->res->minimum_distance = dist_result;
 
-      if (!cdata->req->global)
+      if (false /*!cdata->req->global*/)
```
Comment (davetcoleman) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp]: same

**Comment by jrgnicho** [comment]
>Does not git history achieve this "just in case" scenario? It seems to me keeping it will only confuse future contributors / developers and cause more maintenance hassle

@davetcoleman  thanks for reviewing, indeed git does keep the history.  My argument for keeping **industrial_collision_detection** around is that there may be exclusive functionality there that the constrained_ik planner still needs.  I did my best to "upgrade" **constrained_ik** to the new moveit Distance Query Structures but some code didn't directly translate and so I could've broken something and it would be easier to fix/restore if industrial collision detection was kept.
In order to avoid confusing developers we could add a readme explaining this.

**Comment by jrgnicho** [comment]
@davetcoleman I made the requested changes.

**Comment by davetcoleman** [comment]
> In order to avoid confusing developers we could add a readme explaining this.

Agreed - can you just copy your explanation in your previous comment into the README?

**Comment by jrgnicho** [comment]
@davetcoleman a REAME with an explanation of industrial_collision_detection has been added.

**Review by davetcoleman — APPROVED** [review]

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 3]
```diff
@@ -203,32 +157,37 @@ namespace collision_detection
...
     }
     else
     {
-        dist_threshold = cdata->res->minimum_distance.min_distance;
+        dist_threshold = cdata->res->minimum_distance.distance;
     }
 
     fcl_result.min_distance = dist_threshold;
-    double d = fcl::distance(o1, o2, fcl::DistanceRequest(cdata->req->detailed), fcl_result);
+    double d = fcl::distance(o1, o2, fcl::DistanceRequest(), fcl_result);
```
Comment (Levi-Armstrong) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp]: needs to be fcl::DistanceRequest(cdata->req->enable_nearest_points).

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 4]
```diff
@@ -176,9 +125,14 @@ namespace collision_detection
     double dist_threshold = cdata->req->distance_threshold;
     std::map<std::string, DistanceResultsData>::iterator it1, it2;
 
-    if (!cdata->req->global)
+    /*
+     * FIXME: The old Distance Query Data structures that existed in this package were replaced by their counterparts in the
+     * MoveIt! library.  Unfortunately the legacy code here isn't fully compatible with the new structure types and  so for
+     * now the broken code below has been disabled until the discrepancies get amended.
+     */
+    if (false /*!cdata->req->global*/)
```
Comment (Levi-Armstrong) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp]: cdata->req->type == DistanceRequestType::GLOBAL

File: industrial_collision_detection/src/collision_detection/collision_common.cpp [hunk 5]
```diff
@@ -203,32 +157,37 @@ namespace collision_detection
...
-      cdata->res->minimum_distance.update(dist_result);
+      cdata->res->minimum_distance = dist_result;
 
-      if (!cdata->req->global)
+      /*
+        * FIXME: The old Distance Query Data structures that existed in this package were replaced by their counterparts in the
+        * MoveIt! library.  Unfortunately the legacy code here isn't fully compatible with the new structure types and  so for
+        * now the broken code below has been disabled until the discrepancies get amended.
+        */
+      if (false /*!cdata->req->global*/)
```
Comment (Levi-Armstrong) [inline industrial_collision_detection/src/collision_detection/collision_common.cpp]: Same as above

File: industrial_collision_detection/src/collision_detection/collision_robot_industrial.cpp [hunk 6]
```diff
@@ -157,12 +158,16 @@ void collision_detection::CollisionRobotIndustrial::checkSelfCollisionHelper(con
   manager.manager_->collide(&cd, &collisionCallback);
   if (req.distance)
   {
-    DistanceRequest dreq(false, true, req.group_name, acm);
+    DistanceRequest dreq;
+    dreq.enable_nearest_points = false;
+    dreq.enable_signed_distance = true;
+    dreq.group_name = req.group_name;
+    dreq.acm = acm;
     DistanceResult dres;
```
Comment (Levi-Armstrong) [inline industrial_collision_detection/src/collision_detection/collision_robot_industrial.cpp]: add dreq.type == DistanceRequestType::GLOBAL

File: industrial_collision_detection/src/collision_detection/collision_world_industrial.cpp [hunk 7]
```diff
@@ -122,12 +123,16 @@ void collision_detection::CollisionWorldIndustrial::checkRobotCollisionHelper(co
 
   if (req.distance)
   {
-    DistanceRequest dreq(false, true, req.group_name, acm);
+    DistanceRequest dreq;
+    dreq.enable_nearest_points = false;
+    dreq.enable_signed_distance = true;
+    dreq.group_name = req.group_name;
+    dreq.acm = acm;
     DistanceResult dres;
```
Comment (Levi-Armstrong) [inline industrial_collision_detection/src/collision_detection/collision_world_industrial.cpp]: add dreq.type == DistanceRequestType::GLOBAL

**Review by Levi-Armstrong — CHANGES_REQUESTED** [review]

**Comment by jrgnicho** [comment]
@Levi-Armstrong I've made the requested changes.

**Review by Levi-Armstrong — APPROVED** [review]

**Pull request merged**